spock-core/src/main/java/org/spockframework/compiler/SpecRewriter.java

```diff
@@ -44,6 +47,15 @@ public class SpecRewriter extends AbstractSpecVisitor implements IRewriteResourc
   // https://issues.apache.org/jira/browse/GROOVY-10403
   // needed for groovy-4 compatibility and only available since groovy-4
   private static final java.lang.reflect.Method GET_PLAIN_NODE_REFERENCE = ReflectionUtil.getMethodBySignature(ClassNode.class, "getPlainNodeReference", boolean.class);
+  private static final Set<BlockParseInfo> BLOCK_LISTENER_SUPPORTED_BLOCKS = Collections.unmodifiableSet(EnumSet.of(
```

Comment:
How about making this a method in the enum instead?
Then when adding a new block type it cannot be forgotten to check here, and `AND` can delegate to the previous block like with the other methods.